src/lib.rs

```diff
@@ -181,7 +181,7 @@ pub fn verify_image(
     // so `image_length` is significantly smaller than `u32::MAX`. So, we
     // override the overflow check.
     let image_length_fwords =
-        header.image_length.wrapping_add(BYTES_PER_FWORD_U32 + 1) / BYTES_PER_FWORD_U32;
+        header.image_length.wrapping_add(BYTES_PER_FWORD_U32 - 1) / BYTES_PER_FWORD_U32;
```

Comment:
I think it may be even [even worse](https://rust.godbolt.org/z/hheqv7Gh8) on armv8m which is the RoT target. Even more unfortunate is that bootleby is still pinned to an older rust version where that isn't stable

```
error[E0658]: use of unstable library feature 'int_roundings'
   --> src/lib.rs:184:29
    |
184 |         header.image_length.div_ceil(BYTES_PER_FWORD_U32);
    |                             ^^^^^^^^
    |
    = note: see issue #88581 <https://github.com/rust-lang/rust/issues/88581> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `bootleby` due to previous error
warning: build failed, waiting for other jobs to finish...
```

Size before with `cargo build --release --no-default-features --features target-board-oxide-rot-1`
```
labbott@labbott-desktop2:~/bootleby$ arm-none-eabi-size target/thumbv8m.main-none-eabihf/release/bootleby
   text	   data	    bss	    dec	    hex	filename
   2752	      0	 721988	 724740	  b0f04	target/thumbv8m.main-none-eabihf/release/bootleby
labbott@labbott-desktop2:~/bootleby$ git rev-parse HEAD
3e10edc514b47f31a3f97b2d4821ebe4c1d86115
```

With just a toolchain bump plus the fix
```
labbott@labbott-desktop2:~/bootleby$ arm-none-eabi-size target/thumbv8m.main-none-eabihf/release/bootleby
   text	   data	    bss	    dec	    hex	filename
   2804	      0	 721988	 724792	  b0f38	target/thumbv8m.main-none-eabihf/release/bootleby
```

And after with `div_ceil` plus a toolchain bump
```
labbott@labbott-desktop2:~/bootleby$ arm-none-eabi-size target/thumbv8m.main-none-eabihf/release/bootleby
   text	   data	    bss	    dec	    hex	filename
   2812	      0	 721988	 724800	  b0f40	target/thumbv8m.main-none-eabihf/release/bootleby
```

So the cost of doing a toolchain bump is higher than doing the `div_ceil` fix. I'm torn between saying "no we want our first update to be small" and also saying "shove as much in there as possible so we have fewer updates later".

Reply:
Sounds good, let's take this change as is. I'll keep this conversation open for future us and file an issue about making this change later because I like the intent of making the code easier for us to read.